source/features/limit-commit-title-length.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import onetime from 'onetime';
+import features from '../libs/features';
+
+const getError = onetime<[], HTMLElement>((): HTMLElement =>
+	<p className="note text-red">Title of commit should be less than 70 characters.</p>
+);
+
+function init(): void {
+	const inputs = select.all<HTMLInputElement>([
+		'#commit-summary-input', // Commit title on edit file page
+		'#pull_request_title', // PR title while creating PR
+		'#merge_title_field' // PR merge message field
+	].join(','));
+
+	for (const inputField of inputs) {
+		const validityCallback = (): void => {
+			if (inputField.checkValidity()) {
+				getError().remove();
+				return;
+			}
+
+			inputField.after(getError());
+		};
+
+		inputField.setAttribute('maxlength', '70');
+
+		inputField.addEventListener('input', validityCallback);
```

Comment:
I don’t think we need a custom UI for this. Can you ensure that the browser triggers the regular form validation? It’ll say “the field is longer than allowed” without any additional JS

Actually this would be even better than limiting with maxlength because maxlength makes it hard to edit the field.

Reply:
Yes, GitHub validates all inputs before submitting using the forms `validate` attribute. But I think there is no way for us to trigger the browsers popup by our self.

The popup appears only while trying to submit the form, not while editing the field. For comparison, GitHub shows it's own warning message while typing, but that's not a strict enforcement anyway.

Reply:
>  think there is no way for us to trigger the browsers popup by our self.

The popup naturally comes up when validation is required, when the form is submitted, but I don't know if GitHub is using `novalidate` or something

Reply:
Yes, we can set custom error message using `setCustomValidity()`, but I wanted a warning while the user was typing. The custom validation message only appears when form is being submitted.

If that's too much of code to have for too little use, I'd be happy to drop it.

Reply:
> but I wanted a warning while the user was typing

Not a bad idea, but I wouldn't want to make the area too busy, we already have some small text there from `sync-pr-commit-title`

How about using some `:invalid` style like in `options.css`?

Reply:
![image](https://user-images.githubusercontent.com/37769974/58856496-75d13b00-86c0-11e9-88d7-f1749e3e2edf.png)

😅